src/Microsoft.AspNet.Mvc.Xml/XmlDataContractSerializerInputFormatter.cs

```diff
@@ -19,6 +20,7 @@ namespace Microsoft.AspNet.Mvc.Xml
     public class XmlDataContractSerializerInputFormatter : InputFormatter
     {
         private DataContractSerializerSettings _serializerSettings;
+        private ConcurrentDictionary<Type, object> _serializerCache = new ConcurrentDictionary<Type, object>();
```

Comment:
@yishaigalatzer : should this be static? even though formatters by default are registered as single instance, users can make them per-request too.